src/Storages/ObjectStorage/StorageObjectStorage.h

```diff
@@ -193,10 +211,16 @@ class StorageObjectStorage::Configuration
     /// buckets in S3. If object storage doesn't have any namepaces return empty string.
     virtual std::string getNamespaceType() const { return "namespace"; }
 
-    virtual Path getFullPath() const { return ""; }
+    Path getRawPath() const;
```

Comment:
Add a comment with explanation, what is this paths and when they have difference. Later without context of this PR will be hard to understand difference, I think.